spec/e2e/query_proxy_methods_spec.rb

```diff
@@ -72,6 +72,110 @@ def destroy_called
     end
   end
 
+  describe 'find_or_initialize_by' do
+    context 'with relations chain' do
+      let(:emily)       { Student.create(name: 'Emily') }
+      let(:philosophy)  { Lesson.create(name: 'philosophy') }
+      before do
+        philosophy.students << jimmy
```

Comment:
Optionally you could also do

```ruby
Lesson.create(name: 'philosophy', students: [jimmy])
```

In the `philosophy` block.  Don't know if that would break any tests because this `before` is currently making sure that these `let` variables are referenced before any of the `it` blocks are run